File: app/src/main/java/com/sevtinge/hyperceiler/compat/EzXHelper.java
package com.sevtinge.hyperceiler.compat;

import android.content.Context;

import io.github.lingqiqi5211.ezhooktool.xposed.EzXposed;

/**
 * legacy EzXHelper 的 API 102 兼容影子。
 *
 * <p>只保留模块在用的基础能力；hook 相关的 EzXHelper DSL 请改用 EzHookTool。</p>
 */
public final class EzXHelper {

    private static String sLogTag = "HyperCeiler";
    private static String sToastTag = "HyperCeiler";

    private EzXHelper() {
    }

    public static void initZygote(StartupParam startupParam) {
        // API 102 下模块路径由入口直接提供，这里只做兼容占位。
    }

    public static void initHandleLoadPackage(XC_LoadPackage.LoadPackageParam lpparam) {
        HookRuntime.attachPackage(lpparam.packageName, lpparam.classLoader);
    }

    public static void setLogTag(String tag) {
        sLogTag = tag;
    }

    public static String getLogTag() {
        return sLogTag;
    }

    public static void setToastTag(String tag) {
        sToastTag = tag;
    }

    public static String getToastTag() {
        return sToastTag;
    }

    public static ClassLoader getClassLoader() {
        return HookRuntime.classLoader();
    }

    public static Context getAppContext() {
        Context context = HookRuntime.appContext();
        if (context != null) return context;
        return EzXposed.getAppContextOrNull();
    }

    public static String getHostPackageName() {
        return EzXposed.getPackageName();
    }
}

File: app/src/main/java/com/github/kyuubiran/ezxhelper/EzXHelper.kt
@file:Suppress("unused", "MemberVisibilityCanBePrivate")

package com.github.kyuubiran.ezxhelper

import android.content.Context
import io.github.lingqiqi5211.ezhooktool.xposed.EzXposed

/**
 * EzXHelper 的 API 102 兼容实现。
 */
object EzXHelper {

    /** 当前日志 tag。 */
    var logTag: String = "HyperCeiler"

    /** 当前 toast tag。 */
    var toastTag: String = "HyperCeiler"

    /** 目标进程 application context；尚未就绪时为 null。 */
    val appContext: Context?
        get() = EzXposed.appContextOrNull

    @JvmStatic
    fun getAppContext(): Context? = appContext

    /** 当前目标 ClassLoader。 */
    val classLoader: ClassLoader
        get() = EzXposed.safeClassLoader

    @JvmStatic
    fun getClassLoader(): ClassLoader = classLoader

    /** 当前宿主包名。 */
    val hostPackageName: String
        get() = EzXposed.packageName

    @JvmStatic
    fun getHostPackageName(): String = hostPackageName

    /** 当前进程名。 */
    val processName: String
        get() = EzXposed.processName

    @JvmStatic
    fun setLogTag(tag: String) {
        logTag = tag
    }

    @JvmStatic
    fun setToastTag(tag: String) {
        toastTag = tag
    }

    @JvmStatic
    fun initAppContext(context: Context?) {
        if (context != null) {
            runCatching { EzXposed.initAppContext(context) }
        }
    }

    @JvmStatic
    fun initZygote(startupParam: Any?) {
        // API 102 下由模块入口统一初始化。
    }

    @JvmStatic
    fun initHandleLoadPackage(lpparam: Any?) {
        // API 102 下由 EzHookTool 运行时管理 ClassLoader。
    }
}
